lib/active_model/serializer.rb

```diff
@@ -159,64 +175,66 @@ def self.cache(options = {})
-    end
-
-    # @see ActiveModel::Serializer::Adapter.lookup
-    def self.adapter
-      ActiveModel::Serializer::Adapter.lookup(config.adapter)
-    end
-
     # Used to cache serializer name => serializer class
     # when looked up by Serializer.get_serializer_for.
     def self.serializers_cache
       @serializers_cache ||= ThreadSafe::Cache.new
     end
 
-    # @api private
-    def self.serializer_lookup_chain_for(klass)
-      chain = []
+    # @example
+    #   class AdminAuthorSerializer < ActiveModel::Serializer
+    #     type 'authors'
+    def self.type(type)
+      self._type = type
+    end
 
-      resource_class_name = klass.name.demodulize
-      resource_namespace = klass.name.deconstantize
-      serializer_class_name = "#{resource_class_name}Serializer"
+    def self.link(name, value = nil, &block)
+      _links[name] = block || value
+    end
 
-      chain.push("#{name}::#{serializer_class_name}") if self != ActiveModel::Serializer
-      chain.push("#{resource_namespace}::#{serializer_class_name}")
+    # @api private
+    # names of attribute methods, @see Serializer#attribute
+    def self._attributes
+      serialized_attributes.keys
+    end
 
-      chain
+    # @example
+    #   class AdminAuthorSerializer < ActiveModel::Serializer
+    #     attributes :id, :name, :recent_edits
+    def self.attributes(*attrs)
+      attrs = attrs.first if attrs.first.class == Array
+
+      attrs.each do |attr|
+        attribute(attr)
+      end
     end
 
-    # @api private
-    # Find a serializer from a class and caches the lookup.
-    # Preferentially retuns:
-    #   1. class name appended with "Serializer"
-    #   2. try again with superclass, if present
-    #   3. nil
-    def self.get_serializer_for(klass)
-      serializers_cache.fetch_or_store(klass) do
-        # NOTE(beauby): When we drop 1.9.3 support we can lazify the map for perfs.
-        serializer_class = serializer_lookup_chain_for(klass).map(&:safe_constantize).find { |x| x && x < ActiveModel::Serializer }
+    # @example
+    #   class AdminAuthorSerializer < ActiveModel::Serializer
+    #     attributes :id, :recent_edits
+    #     attribute :name, key: :title
+    #     attribute :name do
+    #       "#{object.first_name} #{object.last_name}"
+    #     end
+    #
+    #     def recent_edits
+    #       object.edits.last(5)
+    #     end
+    # TODO: remove dynamic method definition
+    def self.attribute(attr, options = {}, &block)
+      key = options.fetch(:key, attr)
+      _attributes_keys[attr] = { key: key } if key != attr
 
-        if serializer_class
-          serializer_class
-        elsif klass.superclass
-          get_serializer_for(klass.superclass)
-        end
+      if block_given?
+        serialized_attributes[key] = Attribute.new(->(instance) { instance.instance_eval(&block) })
+      else
+        serialized_attributes[key] = Attribute.new(->(instance) { instance.object.read_attribute_for_serialization(attr) })
```

Comment:
Yeah, I think we talked about it and I agree that at some point it would be nice, but at the moment I think it just adds complexity.